source/beamline/CLS/CLSBeamline.h

```diff
@@ -27,13 +27,33 @@ class CLSBeamline : public AMBeamline
 	/// Destructor.
 	virtual ~CLSBeamline();
 
+	/// function to be called after beamline instantiated, which will finalize the initialization of the CLSBeamline
+	virtual void initializeBeamline();
+
 	/// Returns the beamline's scaler.
 	virtual CLSSIS3820Scaler* scaler() const { return 0; }
+	/// Returns a default XRF detector
+	virtual AMXRFDetector *xrfDetector() const = 0;
+
+signals:
+	/// Signal to notify that the beam availability is changed
+	void beamAvaliabilityChanged(bool beamOn);
+
+protected slots:
+	/// Slot to handle the signals which will affect the beam availability
+	void updateBeamAvailabilityStatus();
 
 protected:
 	/// Protected constructor, for singleton pattern.
 	CLSBeamline(const QString &controlName);
 
+	/// helper function to check whether the beam of a beamline is available or not --- this usually is combined with beamline status PV and/or beamline shutters/valves stauts
+	virtual bool isBeamlineBeamAvailable() = 0;
```

Comment:
It doesn't look like BioXAS has a real implementation for this, `BioXASBeamline::isBeamlineBeamAvailable()` always returns true. If that's the desired default implementation, maybe that could be implemented here?